src/components/ReviewsCustomerPhotos/ReviewsCustomerPhotos.jsx

```diff
@@ -97,40 +96,65 @@ const PreviousImageButton = styled(NextImageButton)`
 class ReviewsCustomerPhotos extends Component {
   constructor(props) {
     super(props);
-    this.state = { translateX: 0 };
+    this.state = { translateX: 0, maxContainerLength: 0 };
+    this.carouselWindow = React.createRef();
   }
 
-  moveRight(reviews, translateX, images, maxLength) {
-    if (translateX === maxLength) {
-      this.setState((PrevState) => ({ translateX: PrevState.translateX - 72 }));
+  componentDidMount() {
+    const { customerImages } = this.props;
+    const imagesTotal = customerImages.length;
+    const containerWidth = this.getContainerWidth();
+    const maxContainerLength = Calc.maxContainerLength(imagesTotal, containerWidth);
+    this.setState({ maxContainerLength });
+  }
+
+  getContainerWidth() {
+    return this.carouselWindow.current.clientWidth;
+  }
+
+  moveRight(imagesTotal, maxContainerLength) {
+    const { translateX } = this.state;
+    const containerWidth = this.getContainerWidth();
+    if ((maxContainerLength - translateX) >= -208) {
+      this.setState((PrevState) => ({
+        translateX: PrevState.translateX + ((maxContainerLength - translateX) - 8), // -8 = padding
+        maxContainerLength: Calc.maxContainerLength(imagesTotal, containerWidth),
+      }));
     } else {
-      this.setState((PrevState) => ({ translateX: PrevState.translateX - 208 }));
+      this.setState((PrevState) => ({
+        translateX: PrevState.translateX - 208,
+        maxContainerLength: Calc.maxContainerLength(imagesTotal, containerWidth),
+      }));
     }
   }
 
-  moveLeft(reviews, translateX, images, maxLength) {
-    if (translateX === maxLength) {
-      this.setState((PrevState) => ({ translateX: PrevState.translateX + 72 }));
+  moveLeft(imagesTotal, maxContainerLength) {
```

Comment:
Is there a code common for both moveLeft and moveRight that could be reused?

Reply:
Thanks for the feedback - iIve moved this into a helper function to calculate the last image position. Thanks!